### ISSUE TITLE
Add RFC7231 Accept header parsing 

### ISSUE THREAD

**Issue by ekroon** [roapi/roapi#80]
I would like to use ROAPI endpoints from a browser. Currently when accessing `/api/tables/{table_name}` endpoint with a browser you get an error like the following: 

`{
"code": 400,
"error": "unsupported_content_type",
"message": "\"text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9\" is not a supported response content type"
}`

How to parse the Accept header is specified in [RFC7231][rfc7231] and [Actix][actix-web] has an implementation for this. How to implement this would be impacted by #76 and if this would be implemented it might make switching harder.

An easy quicfix would be to implement a fallback to `application/json` if the resulting content-type is unknown. 

[actix-web]: https://docs.rs/actix-web/4.0.0-beta.9/actix_web/http/header/struct.Accept.html#method.mime_preference "Actix"
[rfc7231]: https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2 "RFC7231"

**Comment by houqp** [comment]
Good catch @ekroon , did you mean to access the endpoint by loading it directly in a browser or loading it through AJAX call?

Regardless, I agree with you that we should support q factor weighting in request header parsing. I think a fallback to json makes sense when accept header is not specified at all by the client. Otherwise, it might confuse the client.

**Comment by ekroon** [comment]
I mean loading it directly in the browser window.

**Comment by houqp** [comment]
Got it, extending on your use-case, I think it might be an interesting idea to implement a html or xml encoding type. The html encoding could render the query result in a nicely looking web page to help better visualize the result. We can even throw in some js code to make the resulting web page interactive.

**Comment by xezpeleta** [comment]
Hi, I agree with @ekroon 

A fallback to application/json would be enough.

Thanks!!

**Comment by ekroon** [comment]
A full implementation for Accept header parsing will be a bit involved as there is no implementation available for Axum. The correct place to implement this would be hyperium/headers#7 but that appears to going slow. Maybe a port of Rocket or Actix's implementation in a separate crate is a quicker solution. 

Alternatively, supporting the format in the url extension, like `.json`, `.xml`, would be an easy usable solution from browsers too. This would work nicely with a future `.html` version.

**Comment by houqp** [comment]
Thanks @ekroon , I agree with both of your points :) For your first point, on top of what's in rocket and actix, I think we can also take https://github.com/hyperium/headers/pull/70 as a reference to build out the crate.
